### PR TITLE
Bug 1797788: add openshift-tests as a future delegation binary

### PR DESCRIPTION
**Pull request by sallyom**
picking up https://github.com/openshift/origin/pull/24450 to get it merged

/assign @soltysh

### CODE REVIEW

**Comment by smarterclayton** [comment]
/hold

Serial and upgrade are failing

**Comment by soltysh** [comment]
@sallyom it looks like this panic is coming from this PR:
```

	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/openshift/origin/vendor/k8s.io/kubernetes/test/integration/etcd.createTestCRD(0x0, 0x62371e0, 0xc003b02460, 0x0, 0xc000e3e000)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/integration/etcd/server.go:315 +0x388
github.com/openshift/origin/vendor/k8s.io/kubernetes/test/integration/etcd.CreateTestCRDs(0x0, 0x62371e0, 0xc003b02460, 0xc003980d00, 0xc003b02660, 0x4, 0x4)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/integration/etcd/server.go:309 +0x61
github.com/openshift/origin/test/extended/etcd.testEtcd3StoragePath(0x6306bc0, 0xc0039bbd40, 0xc003722e00, 0x62a32e0, 0xc0039b9620)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/test/extended/etcd/etcd_storage_path.go:237 +0x1b1
github.com/openshift/origin/test/extended/etcd.glob..func1.1()
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/test/extended/etcd/etcd_test_runner.go:72 +0x1164
github.com/openshift/origin/vendor/github.com/onsi/ginkgo/internal/leafnodes.(*runner).runSync(0xc003d54e40, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/cmd/openshift-tests-kubernetes/openshift-tests-kubernetes.go:233 +0x15d
github.com/openshift/origin/vendor/github.com/spf13/cobra.(*Command).execute(0xc002482280, 0xc001e85db0, 0x1, 0x1, 0xc002482280, 0xc001e85db0)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/github.com/spf13/cobra/command.go:826 +0x460
github.com/openshift/origin/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc0017e3900, 0x0, 0x6195320, 0x9936e28)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/github.com/spf13/cobra/command.go:914 +0x2fb
github.com/openshift/origin/vendor/github.com/spf13/cobra.(*Command).Execute(...)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/github.com/spf13/cobra/command.go:864
main.main.func1(0xc0017e3900, 0x0, 0x0)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/cmd/openshift-tests-kubernetes/openshift-tests-kubernetes.go:57 +0x9c
main.main()
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/cmd/openshift-tests-kubernetes/openshift-tests-kubernetes.go:58 +0x341

fail [runtime/panic.go:679]: Test Panicked: runtime error: invalid memory address or nil pointer dereference
```
pls double check if you can reproduce this locally.

**Comment by sallyom** [comment]
> pls double check if you can reproduce this locally.

I can, also in master branch though. : (

**Comment by soltysh** [comment]
That's interesting, @deads2k @hexfusion you're aware of something like that?

**Comment by sallyom** [comment]
> That's interesting, @deads2k @hexfusion you're aware of something like that?

This fixes things locally, let's see if CI agrees: https://github.com/openshift/origin/pull/24560/commits/270fc616040d6331c4122fa70fa2ad65b73ec25f  although we're set up for failure again in the future.

**Comment by deads2k** [comment]
/lgtm
/retest

but upgrades need to to green.

**Comment by sallyom** [comment]
_sigh_ since e2e-gcp-upgrade passes for me locally, and because the failed tests aren't consistent, I will run this again

/test e2e-gcp-upgrade

**Comment by sallyom** [comment]
test infrastructure errs

/test e2e-gcp-upgrade

**Comment by sallyom** [comment]
test infrastructure errs

/test e2e-gcp-upgrade

**Comment by sallyom** [comment]
test infrastructure errs

/retest

**Comment by sallyom** [comment]
test infrastructure errs

/retest

**Comment by sallyom** [comment]
@deads2k I'm at a loss, `e2e-gcp-upgrade` does not fail for me when I run it locally.  Any idea?  I'll find it..today.. :crossed_fingers:

**Comment by sallyom** [comment]
openshift-apiserver pod logs:
```
I0312 17:28:16.469495       1 trace.go:116] Trace[501080807]: "Get" url:/apis/image.openshift.io/v1/namespaces/openshift-apiserver/imagestreams/missing,user-agent:openshift-tests-kubernetes/v1.17.1 (linux/amd64) kubernetes/bb0c2f2,client:35.231.70.211 (started: 2020-03-12 17:28:11.469651066 +0000 UTC m=+50.332355233) (total time: 4.999808835s):
```

**Comment by sallyom** [comment]
This 100% succeeds locally:
`openshift-tests run-upgrade all --to-image registry.svc.ci.openshift.org/ocp/release:whatever` 
_sifting through logs..again_

File: cmd/openshift-tests/openshift-tests.go [hunk 1]
```diff
@@ -79,7 +79,11 @@ func getTestBinaries(streams genericclioptions.IOStreams) ([]string, error) {
 			}
 
 			testBinaryPath := filepath.Join(dir, f.Name())
-			if testBinaries.Has(testBinaryPath) {
+			isSymlink, err := evalSymlink(testBinaryPath)
```
[the remark this answers is not in the document]
Comment (sallyom) [reply cmd/openshift-tests/openshift-tests.go]: Yes, I'll figure out where/why the symlink is getting created.  I looked at the Dockerfiles but it's not coming from there :woman_shrugging:

**Pull request merged**